File: app/src/main/java/com/kakeibo/data/ContentResource.java
//package com.kakeibo.data;
//
//
//import androidx.annotation.Nullable;
//
//import com.google.gson.Gson;
//import com.google.gson.JsonSyntaxException;
//
//import java.util.Map;
//
//public class ContentResource {
//    private static final String URL_KEY = "url";
//
//    public final String url;
//
//    public ContentResource(String url) {
//        this.url = url;
//    }
//
//    @Nullable
//    public static ContentResource listFromMap(Map<String, Object> map) {
//        Object url = map.get(URL_KEY);
//        if (url instanceof String) {
//            return new ContentResource((String) url);
//        } else {
//            return null;
//        }
//    }
//
//    @Nullable
//    public static ContentResource fromJsonString(String dataString) {
//        Gson gson = new Gson();
//        try {
//            return gson.fromJson(dataString, ContentResource.class);
//        } catch (JsonSyntaxException e) {
//            return null;
//        }
//    }
//}
//

File: app/src/main/java/com/kakeibo/feature_subscriptions/ContentResource.kt
//package com.kakeibo.data
//
//import com.google.gson.Gson
//import com.google.gson.JsonSyntaxException
//
//class ContentResource(val url: String?) {
//    companion object {
//
//        private const val URL_KEY = "url"
//
//        fun listFromMap(map: Map<String, Any>): ContentResource? {
//            val url = map[URL_KEY]
//            return if (url is String) {
//                ContentResource(url as String?)
//            } else {
//                null
//            }
//        }
//
//        fun fromJsonString(dataString: String?): ContentResource? {
//            val gson = Gson()
//            return try {
//                gson.fromJson(dataString, ContentResource::class.java)
//            } catch (e: JsonSyntaxException) {
//                null
//            }
//        }
//    }
//}